tensorboard/plugins/projector/vz_projector/data.ts

```diff
@@ -528,12 +528,15 @@ namespace vz_projector {
           TSNE_SAMPLE_SIZE
         );
         let labels = new Array(sampledIndices.length);
-        sampledIndices.forEach(
-          (index, i) =>
-            (labels[i] = this.points[index].metadata[
-              superviseColumn
-            ].toString())
-        );
+        sampledIndices.forEach((index, i) => {
+          let label = this.points[index].metadata[superviseColumn];
+          if (!label || !label.toString()) {
+            // if rows don't match up, backfill unknown values.
+            label = 'unknown #' + index.toString();
+            this.points[index].metadata[superviseColumn] = label;
+          }
+          labels[i] = label.toString();
```

Comment:
I am a bit unfamiliar with this code. Can you tell us why you need to cast it to string? What is a type of `label`? Here and 533.

Reply:
label is a [string|number], this casts it to a string.